source/base/quadrature_lib.cc

```diff
@@ -943,6 +943,8 @@ QSorted<dim>::QSorted(const Quadrature<dim> &quad) :
     {
       this->weights[i]           = quad.weight(permutation[i]);
       this->quadrature_points[i] = quad.point(permutation[i]);
+      if (permutation[i] != i)
+        this->is_tensor_product_flag = false;
```

Comment:
Right. But what I'm saying is that here I see that you're setting the flag to false under certain conditions, and I'm left to wonder what the previous value was? Can you add a sentence somewhere before that loop that says that the previous value is what it was in the argument to the constructor? (I suspect that the copy constructor of the base class copies the flag.